test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/ConfigurationManagerTests.cs

```diff
@@ -136,6 +136,54 @@ public void FetchMetadataFailureTest()
             TestUtilities.AssertFailIfErrors(context);
         }
 
+        [Fact]
+        public void BootstrapRefreshIntervalTest()
+        {
+            var context = new CompareContext($"{this}.BootstrapRefreshIntervalTest");
+
+            var documentRetriever = new HttpDocumentRetriever(HttpResponseMessageUtils.SetupHttpClientThatReturns("OpenIdConnectMetadata.json", HttpStatusCode.NotFound));
```

Comment:
Not found is not a transient error. We should not retry if this happens. This means that the URL we are using is wrong.

Reply:
We can have a collection of HTTP status codes that are transient and those that are permanent to decide if a retry is required. 

In addition we should also have a list of .NET exceptions for scenarios like timeouts.

Reply:
Agree we should take into consideration retriable vs non-retriable errors but IMO can wait til a subsequent interation.